client/src/lib/error-telemetry.js

```diff
@@ -0,0 +1,68 @@
+/**
+* The error telemetry module handles sending error logging to the backend.
+*
+* NOTE: we _might_ want in the future to have only one Telemetry class and
+* handle both error and "metrics" telemetry, but let's start with a smaller
+* scope at least for now.
+*/
+const Tail   = require('tail').Tail;
+const fs     = require('fs');
+const logger = require('winston');
+
+const TARGET = '/tmp/test';
+
+class ErrorTelemetry {
+  constructor(app, options) {
+    this.app = app;
+    this.options = options;
+  }
+
+  setup() {
+    logger.info('Setting up error logging...');
+    var loggingFile = this.fileToWatch();
+
+
+    // FIXME: this is wrong, we should probably /just/ retry, but handling this an error
+    // BTW, we should define a way to also report critical errors like those to the backend
```

Comment:
I would expect this mechanism for reporting critical errors to be within the scope of [the error telemetry service API contract](https://issues.jenkins-ci.org/browse/JENKINS-49811)